src/app/pages/storage/volumes/datasets/dataset-form/dataset-form.component.ts

```diff
@@ -672,41 +672,65 @@ export class DatasetFormComponent implements Formconfiguration{
 
-            edit_compression_collection = [{label:`Inherit (${this.parent_dataset.compression.rawvalue})`, value: this.parent_dataset.compression.value}];
+            edit_compression_collection = [{label:`Inherit (${this.parent_dataset.compression.rawvalue})`, value: 'INHERIT'}];
             edit_compression.options = edit_compression_collection.concat(edit_compression.options);
 
-            edit_deduplication_collection = [{label:`Inherit (${this.parent_dataset.deduplication.rawvalue})`, value: this.parent_dataset.deduplication.value}];
+            edit_deduplication_collection = [{label:`Inherit (${this.parent_dataset.deduplication.rawvalue})`, value: 'INHERIT'}];
             edit_deduplication.options = edit_deduplication_collection.concat(edit_deduplication.options);
 
-            edit_exec_collection = [{label:`Inherit (${this.parent_dataset.exec.rawvalue})`, value: this.parent_dataset.exec.value}];
+            edit_exec_collection = [{label:`Inherit (${this.parent_dataset.exec.rawvalue})`, value: 'INHERIT'}];
             edit_exec.options = edit_exec_collection.concat(edit_exec.options);
 
 
-            edit_readonly_collection = [{label:`Inherit (${this.parent_dataset.readonly.rawvalue})`, value: this.parent_dataset.readonly.value}];
+            edit_readonly_collection = [{label:`Inherit (${this.parent_dataset.readonly.rawvalue})`, value: 'INHERIT'}];
             edit_readonly.options = edit_readonly_collection.concat(edit_readonly.options);
 
-            edit_atime_collection = [{label:`Inherit (${this.parent_dataset.atime.rawvalue})`, value: this.parent_dataset.atime.value}];
+            edit_atime_collection = [{label:`Inherit (${this.parent_dataset.atime.rawvalue})`, value: 'INHERIT'}];
             edit_atime.options = edit_atime_collection.concat(edit_atime.options);
 
-            edit_recordsize_collection = [{label:`Inherit`, value: this.parent_dataset.recordsize.value}];
+            edit_recordsize_collection = [{label:`Inherit (${this.parent_dataset.recordsize.value})`, value: 'INHERIT'}];
             edit_recordsize.options = edit_recordsize_collection.concat(edit_recordsize.options);
-            entityForm.formGroup.controls['sync'].setValue(pk_dataset[0].sync.value);
-            if (pk_dataset[0].compression.value === 'GZIP') {
-              entityForm.formGroup.controls['compression'].setValue(pk_dataset[0].compression.value+'-6');
+            let sync_value = pk_dataset[0].sync.value;
+            if (pk_dataset[0].sync.source === 'DEFAULT') {
+              sync_value = 'INHERIT';
             }
-            else{
-              entityForm.formGroup.controls['compression'].setValue(pk_dataset[0].compression.value);
+            entityForm.formGroup.controls['sync'].setValue(sync_value);
 
+            let compression_value = pk_dataset[0].compression.value;
+            if (pk_dataset[0].compression.source === 'INHERITED') {
```

Comment:
I think it should be true if INHERITED or DEFAULT, same for all the others below.